### PR TITLE
Add some docs info

### LINKED ISSUE #3115: Some Docs improvement
- [x]  Explain what is an Authlogic model
- [x]  Add how to generate your token to comment with it 
- [x] Explain the master, stable and unstable branches
- [x] Add CodeClimate introduction on codeclimate.md

### PR DESCRIPTION
**Pull request by stefannibrasil**
Fixes #3115 
@publiclab/reviewers could you give me more tips to add to those docs?

When @milaaraujo and I were reading and studying the code a few weeks ago, we found some space for improvement. Would appreciate if you could give more tips here. For example, we don't know exactly how to explain more about the stable and unstable branches and how to use them.

Thanks!

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

* [ ✔️ ] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `rake test`
* [ ✔️ ] code is in uniquely-named feature branch and has no merge conflicts
* [  ✔️] PR is descriptively titled
* [ ✔️ ] ask `@publiclab/reviewers` for help, in a comment below

> We're happy to help you get this ready -- don't be afraid to ask for help, and **don't be discouraged** if your tests fail at first!

If tests do fail, click on the red `X` to learn why by reading the logs.

Please be sure you've reviewed our contribution guidelines at https://publiclab.org/contributing-to-public-lab-software

We have a loose schedule of reviewing and pulling in changes every Tuesday and Friday, and publishing changes on Fridays. 

Thanks!

### CODE REVIEW

File: doc/TESTING.md [hunk 1]
```diff
@@ -16,23 +16,23 @@ The tests take between 6-12 minutes to run (we're working on shortening this!) s
...
 
 `RAILS_ENV=test bundle exec rake spec:javascript`
 
-...[if you have phantomjs installed](#phantomjs-for-javascript-tests) (see above). 
+...[if you have phantomjs installed](#phantomjs-for-javascript-tests) (see above).
 
 ## Solr tests
 
-Solr (search) tests (generally not recommended!) require [installing the Solr search engine](#solr-search-engine) (see above). Once you've done that, you still need to turn it off in development mode before running tests, with `rake sunspot:solr:stop`. Read more about [this issue here](https://github.com/publiclab/plots2/issues/832#issuecomment-249695309). 
+Solr (search) tests (generally not recommended!) require [installing the Solr search engine](#solr-search-engine) (see above). Once you've done that, you still need to turn it off in development mode before running tests, with `rake sunspot:solr:stop`. Read more about [this issue here](https://github.com/publiclab/plots2/issues/832#issuecomment-249695309).
```
Comment (jywarren) [inline doc/TESTING.md]: We can remove this whole section on Solr - as we've stopped using it! Thank you!!!!

File: doc/TESTING.md [hunk 2]
```diff
@@ -60,10 +60,18 @@ And to run just a single test within a file:
 
 `ruby -I test test/functional/some_file.rb --name=name_of_test`
 
-### Testing mails in development envirmoment
+### Testing mails in development environment
```
Comment (jywarren) [inline doc/TESTING.md]: 😆 thanks for the catch!!!

File: doc/TESTING.md [hunk 3]
```diff
@@ -60,10 +60,18 @@ And to run just a single test within a file:
...
 
-We are using 'letter_opener' gem to open the mails in development enviroment.
-Whenever a email is sent then it will automatically catch by letter_opener and it will open in new window in development enviroment, nothing is to be done to run it, it will be done automatically.
+We are using 'letter_opener' gem to open the mails in development environment.
+Whenever a email is sent then it will automatically catch by letter_opener and it will open in new window in development environment, nothing is to be done to run it, it will be done automatically.
 Same links would work, no modification in links are required.
 It will also show us how our actual mail will look like.
 Mail will be same as actual mail we will get in production.
+
+## Testing branches
```
Comment (jywarren) [inline doc/TESTING.md]: 🎉

File: doc/TESTING.md [hunk 4]
```diff
@@ -60,10 +60,18 @@ And to run just a single test within a file:
...
+
+We have three principle branches: a master, where all tested new features are live,
+a stable and an unstable. Those last two are used to test new code before sending
+them to production.
+If you need to use the stable or the unstable branch,
+please ask if someone else is already using it.
```
Comment (jywarren) [inline doc/TESTING.md]: `...in the chatroom (https://publiclab.org/chat)` would be great!

**Comment by stefannibrasil** [comment]
close #3070

**Comment by stefannibrasil** [comment]
hi @jywarren thanks for the tips, I updated the files with them just now :)

**Comment by stefannibrasil** [comment]
hi @jywarren do you think there s still something to add here? :)

**Comment by jywarren** [comment]
OK! Merging!!! Glad that testing bug got fixed! 👍

**Pull request merged**